tree/ntuple/v7/test/ntuple_evolution.cxx

```diff
@@ -0,0 +1,251 @@
+#include "ntuple_test.hxx"
+
+#include <TInterpreter.h>
+
+#include <functional>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include <unistd.h>
+#include <sys/wait.h>
+
+namespace {
+// These helpers are split into an *Impl function and a macro to check for and propagate fatal failures to the caller.
+
+void WriteOldInForkImpl(std::function<void(void)> old)
+{
+   pid_t pid = fork();
+   if (pid == -1) {
+      FAIL() << "fork() failed";
+   }
+
+   if (pid == 0) {
+      old();
+      if (::testing::Test::HasFatalFailure())
+         exit(EXIT_FAILURE);
+      exit(EXIT_SUCCESS);
+   }
+
+   int wstatus;
+   if (waitpid(pid, &wstatus, 0) == -1) {
+      FAIL() << "waitpid() failed";
+   } else if (!WIFEXITED(wstatus) || WEXITSTATUS(wstatus) != 0) {
+      FAIL() << "child did not exit successfully";
+   }
+}
```

Comment:
This will need an implementation on Windows... I'm hoping @bellenot can (eventually) help here, but not needed before deciding that we really want to go that route for unit testing schema evolution.

Also to be checked on macOS - `fork` and `waitpid` should be POSIX functions, but not sure about the headers...

Reply:
Well, there is no `fork()` nor `waitpid()` on Windows. I implemented something loooong time ago for the XRootD client, so if this is needed, I can try to do something similar (but that will never be like `fork()` anyway)

Reply:
We decided to disable the unit test on Windows for now, we will have integration tests coming that will also test schema evolution on Windows.